src/antares/tsgen/ts_generator.py

```diff
@@ -323,39 +524,32 @@ def generate_time_series(
                                 log[(now + index) % log_size] -= logp[(now + index) % log_size]
                                 logp[(now + index) % log_size] = 0
                     current_available_units += cible_retour
-                    current_planned_outages = cluster.npo_max[day]
+                    current_planned_outages = outage_gen_params.npo_max[day]
 
                 fo_candidates = fo_drawer.draw(current_available_units, day)
                 po_candidates, stock = po_drawer.draw(current_available_units, day, stock)
 
                 # apparent PO is compared to cur_nb_AU, considering stock
-                candidate = po_candidates + stock
-                if 0 <= candidate and candidate <= current_available_units:
-                    po_candidates = candidate
-                    stock = 0
-                if candidate > current_available_units:
-                    po_candidates = current_available_units
-                    stock = candidate - current_available_units
-                if candidate < 0:
-                    po_candidates = 0
-                    stock = candidate
+                po_candidates, stock = self._compare_apparent_PO(current_available_units, po_candidates, stock)
 
+                # params : npo_max / min, po_can, current_planned, stock
```

Comment:
Not sure to understand this comment but it seems it's a residual that you can remove